lib/connection/statement.js

```diff
@@ -420,6 +420,12 @@ function createContextPreExec(
     {
       statementContext.requestId = statementOptions.requestId;
     }
+
+    // SNOW-728803: In QA mode, set resubmitRequest to true to test that sqlText
+    // is not overwritten if a sqlText is already specified by the user
+    if (statementOptions.requestId === 'SNOW-728803-requestId') {
```

Comment:
Only happens in QA mode: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/639/commits/2f6a4abfe6089f78aa8ede769c3159af0ff2a635#diff-9fcac555294d59dbbfef783d617b2d0601df65ca8f061b2bfb6805d4e58c0edfR414

Reply:
@sfc-gh-dprzybysz Is there an existing mock SF I can modify for this test? If not, I can look into making one for it

@sfc-gh-pmotacki I don't think there is a specific reason so it's definitely possible to do what you mentioned where `resubmitRequest` is set the same way for both QA and non-QA mode

```
  // use request id passed by user
  if (statementOptions.requestId) {
    statementContext.requestId = statementOptions.requestId;
    statementContext.resubmitRequest = true;
  }
  else {
    // use a random uuid for the statement request id
    statementContext.requestId = uuidv4();
  }
```

Does this look ok?

Reply:
I've remove the test code in production